include/roboteam_ai/world/Field.h

```diff
@@ -1,58 +1,264 @@
+    std::unordered_map<std::string, FieldArcName> CONVERT_TO_FIELD_ARC_NAME = {
+        {"center_circle", CENTER_CIRCLE},
+    };
+
+private:
+    static std::mutex fieldMutex; // Prevents situations where the field state is changed and read at the same time.
+    static Field field; // Stores the field state as a Singleton variable.
+
+    std::optional<double> fieldValues[NUMBER_FIELD_VALUE_NAMES]; // Stores all the constant of the field (lengths, widths, positions)
+    std::optional<FieldLineSegment> fieldLines[NUMBER_FIELD_LINE_NAMES]; // Stores all the lines of the field
+    std::optional<Vector2> fieldVectors[NUMBER_FIELD_VECTOR_NAMES]; // Stores all positions of the field
+    std::optional<FieldArc> fieldArcs[NUMBER_FIELD_ARC_NAMES]; // Stores all the arcs of the field
+
+public:
+    /**
+     * Constructor that creates an unitialized Field
+     */
+    Field() = default;
+
+    /**
+     * Constructor that converts a protobuf message into a Field Message object.
+     * @param sslFieldSize The corresponding protobuf message.
+     */
+    Field(proto::SSL_GeometryFieldSize sslFieldSize);
+
+    /**
+     * Access the field state (using the Singleton pattern).
+     * @return The field state object
+     */
+    static Field get_field();
+
+    /**
+     * Set the field state (using the Singleton pattern).
+     * @param field The new field state
+     */
+    static void set_field(Field _field);
+
+    /**
+     * Get a value/constant about the field. All values are measured in SI standard units, so lengths/distances/widths
+     * are measured in meters.
+     * @param valueName The field value name attribute of which we want to get the value.
+     * @return The corresponding field value.
+     */
+    double operator[](FieldValueName arcName) const;
+
+    /**
+     * Get one of the lines of the field.
+     * @param lineName The field line name which we want to get.
+     * @return The corresponding field line.
+     */
+    FieldLineSegment operator[](FieldLineName lineName) const;
+
+    /**
+     * Get one of the arcs of the field.
+     * @param arcName The field arc name which we want to get.
+     * @return The corresponding field arc.
+     */
+     FieldArc operator[](FieldArcName arcName) const;
+
+    /**
+     * Get one of the positions of the field.
+     * @param vectorName The field vector name of which we want to know the position.
+     * @return The corresponding field vector which points to this position.
+     */
+    Vector2 operator[](FieldVectorName arcName) const;
+
+    /**
+     * Get all the lines of the field
+     * @return A map which contains all field lines
+     */
+    const std::optional<FieldLineSegment>* getField_lines();
+
+private:
+    /**
+     * Convert a float measured in millimeters to meters (is needed, because proto message contains values measured in
+     * millimeters).
+     */
+    static float mm_to_m(float scalar);
```

Comment:
the mm_to_m is also available in utils I believe

Reply:
I will do this in later pull request. This clone was not introduced by this pull request, it was already inside the code.